net/socket/socket.c

```diff
@@ -113,23 +113,8 @@ int psock_socket(int domain, int type, int protocol, FAR struct socket *psock)
        */
 
       ret = g_usrsock_sockif.si_setup(psock, protocol);
-      if (ret == -ENETDOWN)
```

Comment:
But does this break the dual usrsock / nuttx tcp-stack use-case? Device where connectivity can be configured to use usrsock or nuttx tcp-stack?

Reply:
You mean runtime switch between usrsock and real tcp/ip stack instead of compile decision?

Reply:
Right, when usrsock daemon not running, try nuttx tcp-stack instead.

Reply:
If so, we need revert this patch, but adjust usrsocktest too. Actually, I think many testcase in usrsocktest is too strict to verfiy the errno must equal some specific value. A little internal modification like this will break the testcase.

Reply:
I agree, usrsocktest errno checking should be more relaxed and allow different error values. Specific error values should only be checked when they are really expected.